en/guide/error-handling.md

```diff
@@ -42,6 +42,49 @@ app.get('/', function (req, res, next) {
 })
 ```
 
+Starting with Express 5, route handlers and middleware that return a Promise will call 
+`next(value)` automatically when they reject or throw an error.  For example:
+
+```js
+app.get('/user/:id', function (req, res, next) {
+  return getUserById(req.params.id).then(function (user) {
+    res.send(user)
+  })
+})
+```
+
+If `getUserById` throws an error or rejects, `next` will be called with either the thrown error or the rejected value.  
+If no rejected value is provided, `next` will be called with the string `Rejected promise`.
+
+If you want to handle the error before passing it along to `next`, you can use a try/catch block (for functions using async/await) or add a `.catch` to your Promise (or an onRejected callback to `.then`).  For example:
+
+```js
+app.get('/user/:id', function (req, res, next) {
```

Comment:
I think we would do better with `async`/`await` in most examples.  In this case, this might be more simple for beginners to understand:

```
app.get('/user/:id', async function (req, res, next)
  const user = await getUserById(req.params.id)
  res.send(user)
})
```

The nice thing about async await for these examples is you can drop a bunch of things which make the code harder to read and might confuse folks just learning where `async`/`await` is more common.  Thoughts?

Reply:
agree @wesleytodd and I'll update.  What's the lowest supported node version for Express 5? 

While async/await is what most of the community is using at this point, given the versions of node that Express 5 supports, I wanted to make sure some examples didn't include async/await syntax so people didn't think they _had_ to use async/await.

Reply:
Agree about async/await being more common and I'll make the update.

Reply:
As for the minimum version, it is no lower than 4, but likely going to be either 6 or 8. It hasn't been set in stone yet, but the guide pages today don't even support all the versions that 4 does due to the usage of arrow functions.

Reply:
> As for the minimum version, it is no lower than 4, but likely going to be either 6 or 8. It hasn't been set in stone yet, but the guide pages today don't even support all the versions that 4 does due to the usage of arrow functions.

Once the minimum node version for express 5 is set, I think it'd be great to go over the entire site and update it to include the most modern syntax possible.

Reply:
Here's an existing issue https://github.com/expressjs/expressjs.com/issues/470 related to updating to es6 syntax.  Once we  figure out the minimum supported version, someone can make the syntax consistent across the site